### ISSUE TITLE
Select Menu: popup width should always be >= dropdown width

### ISSUE THREAD

**Issue by fbuchinger** [fnagel/jquery-ui#251]
Hi,

we are using SelectMenu as a replacement for standard <SELECT>s, because of IE 7/8's bad habit to crop exceeding <OPTION> texts in sized select fields. So far we are quite happy with it, the only downside is that SelectMenu only allows little control over the popup sizing. 

Via the menuWidth parameter, I can only set a fixed size for the popup, so longer texts will be wrapped to multiple lines. When I set the `menuWidth` to `auto`, the longer texts render ok, therefore shorter texts lead to a shrinked popup. I've demonstrated this in a fiddle: 

http://jsfiddle.net/9GNpg/

In short, the width of the popup should always be at least the width of the dropdown.

regards,

Franz

**Comment by fnagel** [comment]
Im sorry but you're using another script by filament group. Ive only maintain the Selectmenu Widget and give some support in my free time. Please see the jQuery UI Wiki as FG decided to contribute this plugin to jQuery UI.

Blogpost: http://www.filamentgroup.com/lab/jquery_ipod_style_and_flyout_menus//
jQuery UI Wiki: http://wiki.jqueryui.com/w/page/12137997/Menu

**Comment by fnagel** [comment]
Any feedback on this issue?

**Comment by fbuchinger** [comment]
Sorry for the silence... vacation time ;)

Um, something wrong must have happened with my fiddle, because it shows something completely different. Here is the correct one: http://jsfiddle.net/fbuchinger/QakXx/9/

My problem boils down to this: the select menu widgets on my page are sized by a layout grid and fed by the server with `<options>` of variable length. often, the option text exceeds the length of the selectmenu, so the selectmenu popup should be longer than the widget itself, up to a given max limit. I accomplish this with the combination of `width` and `menuWidth` parameters. However, the option text might also be much shorter than the widget width. in this case, the popup should be exact the widget width, not longer (like in the fiddle). 

So I was thinking about an `maxMenuWidth` parameter, that enlarges the popup ONLY if the content requires it. 
Or are there other options?

**Comment by fnagel** [comment]
No, there are no other options so far. Perhaps you could try not to specify an width and just use CSS, perhaps with max-width. But Im not sure if this will work. You should give it a try!

Anyway, if this wont work, please open a new issue to discuss the width calculation in general. There were multiple issues (please use the search, perhaps your usecase was already discussed!) about width calculation and if we change something here, we should be sure to match most usecases.

**Comment by fnagel** [comment]
Any feedback on this issue?

**Comment by fbuchinger** [comment]
I didn't dig into the problem any further because our client was satisfied with the `width` / `menuWidth` approach. Regarding the width calculation algorithm I'd look how this is implemented elsewhere in jQuery UI, e.g. the 'autoHeight' option of the accordion widget (http://jqueryui.com/demos/accordion/#option-autoHeight). .
